c/validator.h

```diff
@@ -202,6 +231,22 @@ void _csal_parent_path(uint8_t key[32], uint8_t height) {
     _csal_copy_bits(key, height + 1);
   }
 }
+void _csal_merge(blake2b_state *blake2b_ctx,
+                 const uint8_t zero_value[32],
+                 const uint8_t lhs[32],
+                 const uint8_t rhs[32],
+                 uint8_t output[32]) {
+  if (memcmp(lhs, zero_value, 32) == 0) {
```

Comment:
I think it's better we use a loop to test if any of the values in the 32-byte hash is zero, than passing in a weird `zero_value` array.

Reply:
You are right. 

I was thought the code would be more clean.